bin/node/runtime/src/lib.rs

```diff
@@ -564,8 +565,8 @@ impl_runtime_apis! {
 		}
 
 		fn finalized_block() -> (u64, sp_bridge_eth_poa::H256) {
-			let finalized_block = BridgeRialto::finalized_block();
-			(finalized_block.number, finalized_block.hash)
+			let finalized_block = BridgeRialto::best_finalized_header();
+			(finalized_block.number, finalized_block.compute_hash())
```

Comment:
`compute_hash` might be expensive, I think the bridge should be able to return that cheaply as well.
Maybe it's not super important in that context (`RuntimeApi`), but the fact that the relay needs this information indicates that it might be worth to extend the `MinimalHeaderChain` to provide that as well.